### PR TITLE
Add timestamps on request and responses

### PR DESCRIPTION
**Pull request by fornwall**

### CODE REVIEW

**Review by nakulbansal1609 — COMMENTED** [review]
The change for timestamp format is fine.

File: http-types-schema.json [hunk 1]
```diff
@@ -86,6 +92,12 @@
                 { "type": "array", "items": { "type": "string" } }
               ]
           }
+        },
+        "timestamp": {
+          "title": "HttpResponseTimestamp",
+          "description": "The time at which the HTTP response was completed on the server side in ISO 8601 format, as in '2018-11-13T20:20:39+00:00'.",
+          "type": "string",
+          "format": "date-time"
```
Comment (ksaaskil) [inline http-types-schema.json:100]: I assume it's never possible in practice to have both request and response timestamps in the object? Just wondering if we could simplify things by just having a single timestamp...

File: README.md [hunk 2]
```diff
@@ -23,15 +23,17 @@ A single request and response pair is represented as shown in the below example:
       "user-agent": "Mozilla/5.0 (pc-x86_64-linux-gnu) Siege/3.0.8"
     },
     "pathname": "/user/repos",
-    "query": { "param": "value" }
+    "query": { "param": "value" },
+    "timestamp": "2018-11-13T20:20:39+01:00"
```
Comment (ksaaskil) [inline README.md:27]: Would milliseconds be allowed by the `date-time` format?

**Review by ksaaskil — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (fornwall) [reply README.md:27]: Yes - the ISO 8601 format (which date-time means) allows arbitrary precision, as in `2018-11-13T20:20:39+01:00.1234243234521345124`.

In practice implementations will round it, but I think microsecond should be safe across implementations.

**Comment by MariaMalitckaya** [comment]
Looks good for me!

File: http-types-schema.json (hunk 1 above)
Comment (fornwall) [reply http-types-schema.json:100]: I think what we ourselves need and get here (from things like server logs) is the response timestamp, when the response was done.

I've added the request timestamp, when the request was initiated, to (1) be able to measure time taken between request initiated and response sent and (2) we might want to represent requests that caused a timeout (no response received). But if there is no immediate need for it I have no problem removing `request.timestamp` and revisiting it once needed - what do you think?

**Review by nakulbansal1609 — APPROVED** [review]

**Comment by MariaMalitckaya** [comment]
Sorry, I closed that by mistake=)

File: http-types-schema.json (hunk 1 above)
Comment (fornwall) [reply http-types-schema.json:100]: We're adding both fields for now.

**Pull request merged**